File: Amazon.Lambda/Felipearpa.Tyche.Pool/src/Application/GetPoolMatchGamblerBets.fs
namespace Felipearpa.Tyche.Pool.Application

open Felipearpa.Tyche.Pool.Domain
open Felipearpa.Type

type GetPoolMatchGamblerBets(poolGamblerBetRepository: IPoolGamblerBetRepository) =

    member this.ExecuteAsync(poolId: Ulid, matchId: Ulid, next: string option) =
        poolGamblerBetRepository.GetPoolMatchGamblerBetsAsync(poolId, matchId, next)

File: Android/data/bet/src/main/java/com/felipearpa/tyche/data/bet/application/GetPoolMatchGamblerBets.kt
package com.felipearpa.tyche.data.bet.application

import com.felipearpa.tyche.data.bet.domain.PoolGamblerBetRepository

class GetPoolMatchGamblerBets(private val poolGamblerBetRepository: PoolGamblerBetRepository) {
    suspend fun execute(
        poolId: String,
        matchId: String,
        next: String? = null,
    ) =
        poolGamblerBetRepository.getPoolMatchGamblerBets(
            poolId = poolId,
            matchId = matchId,
            next = next,
        )
}
